modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,12 +48,31 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeRealignPeerIds: Option[NonEmptySet[PeerId]] = none
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
-
       private val logger = Slf4jLogger.getLogger[F]
 
+      def realignToMajority(lastSnapshot: Hashed[GlobalIncrementalSnapshot]): F[Unit] =
+        maybeRealignPeerIds.traverse_ { realignPeers =>
+          for {
+            peers <- realignPeers.toNonEmptyList
+              .traverse(globalL0ClusterStorage.getPeer)
+              .map(_.toList.flatten)
+
+            majorityHash <- peers
+              .parTraverseN(10)(p => l0GlobalSnapshotClient.getHash(lastSnapshot.ordinal).run(p))
+              .map(_.flatten)
+              .map(pickMajority[List, Hash])
+
+            isAligned = majorityHash.fold(false)(lastSnapshot.hash === _)
+
+            maybeL0Peers = SortedSet.from(peers).toNes
+            _ <- maybeL0Peers.traverse_(globalL0ClusterStorage.setPeers).unlessA(isAligned)
```

Comment:
I think that if you first find the majority and then download the majority snapshot then the peers will be automatically properly set by the rediscovery process. If we don't like the fact that the l0 peer rediscovery process is a periodic maybe we can synchronously rediscover peers after every batch of snapshots or every snapshot?

Reply:
I believe this should be covered by other comments now.